internal/server/v1/warden/handler_test.go

```diff
@@ -0,0 +1,258 @@
+		}}, nil)
+
+		doer := mocks.NewDoer(t)
+		doer.EXPECT().Do(mock.Anything).Return(&http.Response{
+			StatusCode: http.StatusOK,
+			Body: io.NopCloser(bytes.NewBufferString(`{
+				"success": true,
+				"data": {
+					"name": "data_fabric",
+					"created_at": "2023-08-25T03:52:13.548Z",
+					"updated_at": "2023-08-25T03:52:13.548Z",
+					"owner_id": 433,
+					"parent_team_identifier": "2079834a-05c4-420d-bfc8-44b934adea9f",
+					"identifier": "b5aea046-dab3-4dac-b1ea-e1eef423226b",
+					"product_group_name": "data_engineering",
+					"product_group_id": "2079834a-05c4-420d-bfc8-44b934adea9f",
+					"labels": null,
+					"short_code": "T394"
+				},
+				"errors": []
+			}`)),
+		}, nil)
+
+		r := chi.NewRouter()
+		r.Use(reqctx.WithRequestCtx())
+		r.Route("/dex/warden", Routes(shieldClient, doer))
+
+		req, err := http.NewRequestWithContext(context.TODO(), http.MethodPatch, "/dex/warden/groups/e38527ee-a8cd-40f9-98a7-1f0bbd20909f/metadata", bytes.NewBufferString(`{"warden_team_id": "123"}`))
+		assert.NoError(t, err)
+
+		resp := httptest.NewRecorder()
+		r.ServeHTTP(resp, req)
+
+		assert.Equal(t, http.StatusOK, resp.Code)
+		assert.JSONEq(t, `{
+			"privacy": "public",
+			"product-group-id": "2079834a-05c4-420d-bfc8-44b934adea9f",
+			"team-id": "b5aea046-dab3-4dac-b1ea-e1eef423226b"
+		}`, resp.Body.String())
+	})
+
+	t.Run("should return 500 when warden_team_id is invalid", func(t *testing.T) {
+		shieldClient := shareMocks.NewShieldServiceClient(t)
+
+		doer := mocks.NewDoer(t)
+		doer.EXPECT().Do(mock.Anything).Return(&http.Response{
+			StatusCode: http.StatusOK,
+			Body: io.NopCloser(bytes.NewBufferString(`{
+				"success": false,
+				"data": {
+					"name": "data_fabric",
+					"created_at": "2023-08-25T03:52:13.548Z",
+					"updated_at": "2023-08-25T03:52:13.548Z",
+					"owner_id": 433,
+					"parent_team_identifier": "2079834a-05c4-420d-bfc8-44b934adea9f",
+					"identifier": "b5aea046-dab3-4dac-b1ea-e1eef423226b",
+					"product_group_name": "data_engineering",
+					"product_group_id": "2079834a-05c4-420d-bfc8-44b934adea9f",
+					"labels": null,
+					"short_code": "T394"
+				},
+				"errors": []
+			}`)),
+		}, nil)
+
+		r := chi.NewRouter()
+		r.Use(reqctx.WithRequestCtx())
+		r.Route("/dex/warden", Routes(shieldClient, doer))
+
+		req, err := http.NewRequestWithContext(context.TODO(), http.MethodPatch, "/dex/warden/groups/e38527ee-a8cd-40f9-98a7-1f0bbd20909f/metadata", bytes.NewBufferString(`{"warden_team_id": "123"}`))
+		assert.NoError(t, err)
+
+		resp := httptest.NewRecorder()
+		r.ServeHTTP(resp, req)
+
+		assert.Equal(t, http.StatusInternalServerError, resp.Code)
+	})
+
+	t.Run("should return error when warden_team_id is empty string", func(t *testing.T) {
+		shieldClient := shareMocks.NewShieldServiceClient(t)
+		doer := mocks.NewDoer(t)
+
+		r := chi.NewRouter()
+		r.Use(reqctx.WithRequestCtx())
+		r.Route("/dex/warden", Routes(shieldClient, doer))
```

Comment:
this is setup outside the module and we dont need it.

we should be able to skip this step and do it like the following
https://github.com/goto/dex/blob/main/internal/server/v1/optimus/routes_test.go#L51-L55